### ISSUE TITLE
Pose prediction biased toward upright — fallen objects still estimated as standing

### ISSUE THREAD

**Issue by Arrow-gh** [Omni6DPose/GenPose2#25]
Hi, thanks for the great work on GenPose++! We've been using it for 6DoF pose estimation in robotic manipulation and it works very well for upright tabletop objects.

However, when a bottle is tilted to ~90° (lying on the table), the predicted Y-axis still points upward as if the object were standing — the model doesn't recognize that it has fallen over. Slight tilts (~45°) are estimated correctly. Thin axes = ground truth, thick axes = prediction:

<img width="1280" height="960" alt="Image" src="https://github.com/user-attachments/assets/5662ff4a-1475-43d7-9c04-9ead8f87b44c" />

In addition to testing on bottles, we also tested other common objects and observed the same problem. Have you observed this, or do you have suggestions for improving robustness to non-upright orientations?

**Comment by Arrow-gh** [comment]
I tried generating a dataset of 3,600 non-upright objects to fine-tune ScoreNet and PointNet, but the estimation accuracy became worse after fine-tuning. It even failed to correctly recognize upright objects. Is this because the dataset is too small, or should I train the model from scratch instead?

**Comment by Jiyao06** [comment]
Thank you for the feedback. We believe this behavior is primarily caused by bias in the pose distribution of the training data. You may also try the `tracking mode`, where the pose estimate from the previous frame is used to initialize the optimization for the current frame. In sequential settings, this often provides better temporal consistency and improves the stability of the predictions. For implementation details, please refer to the `Real-time camera stream inference` section.
